### DISCUSSION TITLE
ERROR:   No invertible representation for getDualRay

### DISCUSSION THREAD

**Discussion by SeanByrne08** [spine-tools/SpineOpt.jl#808] [Ideas]
Hi, 

I have been encountering the error 'No invertible representation for getDualRay'. An error doesn't get flagged in SpineOpt but at the end I get this error and the result is returned infeasible. When I run my code for less than roughly 3 months it runs without a problem, it is when I increase the model duration that I get this error. This has started happening since I added different stochastic scenarios and somewhat increased the complexity. Could this be due to computational power?

Any suggestions would be greatly appreciated! Thank you!
Sean Byrne 

The results from SpineOpt are below:

Window 1: 2000-01-01T00:00~>2000-04-30T23:00
Optimizing model EV flexibility...Running HiGHS 1.5.3 [date: 1970-01-01, git hash: 45a127b78]
Copyright (c) 2023 HiGHS under MIT licence terms
Presolving model
151268 rows, 173347 cols, 375420 nonzeros
Problem status detected on presolve: Infeasible
Model   status      : Infeasible
Objective value     :  0.0000000000e+00
HiGHS run time      :          0.45
ERROR:   No invertible representation for getDualRay
  2.368044 seconds (2.99 M allocations: 275.938 MiB, 12.90% gc time)
model is infeasible - if conflicting constraints can be identified, they will be reported below
[ Info: The optimizer HiGHS.Optimizer does not support `compute_conflict!`
Writing report to sqlite:///C:\Users\Sean\OneDrive\Desktop\Project Model V2H Stochastic 2\.spinetoolbox\items\output_data\Output data.sqlite?spinedbfilter=cfg%3Aexecution%3A%7B%22execution_item%22%3A+%22Run+SpineOpt%22%2C+%22scenarios%22%3A+%5B%5D%2C+%22timestamp%22%3A+%222023-11-24T15%3A01%3A28%22%7D...  0.146662 seconds (2.20 k allocations: 949.977 KiB)

Execution complete. Started at 2023-11-24T15:01:46.522, ended at 2023-11-24T15:05:36.660, elapsed time: 3 minutes, 50 seconds, 138 milliseconds
A JuMP Model
Minimization problem with:
Variables: 287430
Objective function type: JuMP.AffExpr
`JuMP.AffExpr`-in-`MathOptInterface.EqualTo{Float64}`: 130635 constraints
`JuMP.AffExpr`-in-`MathOptInterface.LessThan{Float64}`: 174180 constraints
`JuMP.VariableRef`-in-`MathOptInterface.EqualTo{Float64}`: 1 constraint
`JuMP.VariableRef`-in-`MathOptInterface.GreaterThan{Float64}`: 261299 constraints
Model mode: AUTOMATIC
CachingOptimizer state: ATTACHED_OPTIMIZER
Solver name: HiGHS

**Comment by DillonJ** [comment]
Hi @SeanByrne08,

It looks like your model is infeasible. To diagnose the issue, check that any nodal demands can be satisfied and try adding `node_slack_penalties` to nodes you suspect might be causing a problem.

If you have used any `fix_` parameters, check they aren't causing an infeasibility.

Other things you can try are to run the model for a shorter period of time to try and diagnose the issue.

**Comment by SeanByrne08** [comment]
Thanks Jody, I found the problem!
